### PR TITLE
Require external dispatcher on WS JS backend.

### PR DESCRIPTION
**Pull request by rpiaggio**
When using `Dispacher[F].use`, the dispatcher would be closed as soon as the WS connected, thus unable to further process callbacks.

We now require the `Dispatcher` to passed externally upon creation of the backend instance.

### CODE REVIEW

File: scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala [hunk 1]
```diff
@@ -34,61 +35,60 @@ final class WebSocketJSBackend[F[_]: Async: Logger] extends WebSocketBackend[F]
       isOpen     <- Ref[F].of(false)
       isErrored  <- Ref[F].of(false)
       connection <-
-        Dispatcher[F].use { dispatcher =>
```
Comment (cquiroz) [inline scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala:37]: Ok, I think we had some discussion whether to pass the dispatcher or create it in place but it was never clear the reasons for one or the other

File: scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala [hunk 2]
```diff
@@ -34,61 +35,60 @@ final class WebSocketJSBackend[F[_]: Async: Logger] extends WebSocketBackend[F]
...
+              } yield ()
+            dispatcher.unsafeRunAndForget(close)
           }
         }
     } yield connection
 }
 
 object WebSocketJSBackend {
-  def apply[F[_]: Async: Logger]: WebSocketJSBackend[F] = new WebSocketJSBackend[F]
+  def apply[F[_]: Async: Logger](dispatcher: Dispatcher[F]): WebSocketJSBackend[F] =
```
Comment (cquiroz) [inline scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala:90]: Will this be created on `explore` then?

**Review by cquiroz — APPROVED** [review]
LGTM

File: scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala (hunk 2 above)
Comment (rpiaggio) [reply scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala:90]: Yes. Apps will probably need to keep a global dispatcher, which can be passed here.

**Pull request merged**